### ISSUE TITLE
[Feature request] Support for CSS snippets

### ISSUE THREAD

**Issue by Ver3** [sheabunge/code-snippets#57]
Please consider support for CSS snippets.

### Use case

– Support for both PHP and CSS snippets wil eliminate the use for a child theme.
– No worries about losing custom CSS when updating theme’s.
– Using Export/Import, a lot of time will be saved on new projects.

### I know you must have considered CSS support in the past
I see more people asking for this. I hope the time has come to reconsider. I’ll gladly pay a reasonable yearly fee for a pro version that has this built in. Especially when the CSS is also checked for error and/or has a ‘clean-up-function’.

### If you do consider doing this, count me in as a zealous beta tester!

Ps. Thank you so much for creating this Shea. It’s in all my projects by default.

**Comment by volodymyrpekh** [comment]
@Ver3 Why child theme is not a good solution for you?

**Comment by Ver3** [comment]
@volodymyrpekh I just added my main reason, which answers your question: "Easily turn snippets on/off to check where a problem might be."

I REALY love this feature. Also, I might have OCD and like to keep thing clean and organised. Code snippets helps me do that with PHP. It would be awesome to have the same functionality for CSS snippets. Doubly so because it check the code and make it ex- and importable for other projects.

Hope this answers you question. Feel free to ask more.

**Comment by Ver3** [comment]
@sheabunge As promised, [your reaction from WP Repo](https://wordpress.org/support/topic/feature-request-support-for-css-snippets/) reposted here. Good news indeed!

> "CSS snippets sound like a fantastic idea. I am definitely planning on adding better support for them when I am able to devote the time to creating this feature. Thank you for the suggestions!"

As stated there, I'd really love to be a beta tester on this. Cheers!

**Comment by Ver3** [comment]
Any progress on the addition of CSS Snippets? I'll be happy to help out by beta testing.

**Comment by sheabunge** [comment]
Okay, I'm taking a look and how I might do this, and I have a few questions:

How should CSS snippets be created? Perhaps when adding a new snippet, the user can choose whether it is a CSS or PHP snippet, which then changes the code editor and the way it is run? 

Should CSS and PHP snippets be distinct types of snippets? i.e. a snippet either has CSS code _or_ PHP code, but always only has one block of code.

Once a CSS/PHP snippet has been created, is it locked to that type? Or should it be possible to switch types while editing a snippet? If so, what happens to the existing PHP/CSS code?

I'd love some input on how this might look!

**Comment by gitblair** [comment]
I like the possibility of being able to add a descriptive note for each CSS snippet in the admin, while keeping my CSS notes out of the CSS stylesheet. 

Would all the CSS snippets be concatenated into a single stylesheet?

**Comment by sheabunge** [comment]
@gitblair Yes, that's the idea. I'm planning on there being a generated stylesheet for the admin area and the site-front end, which contains all CSS snippets for that scope.

My questions more related to the experience of creating and editing CSS snippets in the administration.

**Comment by gitblair** [comment]
That sounds great.

As to your questions,

You should start distinctly out knowing you're creating a CSS snippet or PHP snippet and the plugin runs that particular editor for containing specifically either CSS or PHP code. Once created, I can't imagine wanting to switch from one type to another. It'd be just as easy from a user standpoint to just create a new snippet of the type you want.

If all the CSS and PHP snippets are listed on a single page (as opposed to a tabbed interface, I'd suggest color coding the type.)

If a tabbed interface is used, it should be an option to list all on one page.

**Comment by Ver3** [comment]
*How should CSS snippets be created? Perhaps when adding a new snippet, the user can choose whether it is a CSS or PHP snippet, which then changes the code editor and the way it is run?*

### Suggestions

- Rename "Add new" in WP-Admin menu to "Add new PHP".
- Add "Add new CSS" to the same menu.
- Rename "Add new"-button in main snippets panel to "Add new PHP".
- To the right of this button add "Add new CSS"-button.

*Should CSS and PHP snippets be distinct types of snippets? i.e. a snippet either has CSS code or PHP code, but always only has one block of code.*

### Suggestions

- Add CSS-snippets to current list but use a different color for both title and toggle of CSS-snippets. DOing it this way it's easier to export/import all.
- Add two buttons above the snippets list: [Show PHP] and [Show CSS] both toggle all their respective snipptes on/off from view. This way it's very easy to display only PHP-snippts or CSS-snippets.

*Once a CSS/PHP snippet has been created, is it locked to that type? Or should it be possible to switch types while editing a snippet? If so, what happens to the existing PHP/CSS code?*

### Suggestions

- Locked as to prevent confusion.

Hope this helps, call me if you have any questions. Ps. Consider this for your pro version (a topic I'd like to discuss with you when you have time)

**Comment by waynedpj** [comment]
thanks @sheabunge.

could this be easily extended to allow HTML and Javascript snippets as well?

thanks again.

**Comment by sheabunge** [comment]
Hi @waynedpj,

The plan is to use this system to offer all four types of snippets:

- PHP snippets which can be run globally, or restricted to the front-end site or admin area
- CSS snippets which can be loaded in the administration area or site front-end
- JavaScript snippets which can be loaded in the `<head>` section or at the end of the `<body>` section of the site front-end
- HTML snippets which can be used in posts through either a shortcode or a Gutenberg block
